### PR TITLE
feat(API): Club Group Invitation Link API

### PR DESCRIPTION
**Pull request by at-wr**
Still a WIP

Related issue: #485

### CODE REVIEW

**Comment by qwerzl** [comment]
Is it finished? If not, turn it into a draft and request a review from me once completed.

**Comment by at-wr** [comment]
I guess that's it, might be enough for the API of this feature

File: db/schema.prisma [hunk 1]
```diff
@@ -55,17 +56,29 @@ model LeaveRequest {
...
   // Relations
   user User @relation(fields: [rateBy], references: [clerkUserId])
   club Club @relation(fields: [clubId], references: [id])
 
   @@unique([id, clubId, rateBy, rateScope]) // One student can vote only once in one semester
-}
\ No newline at end of file
+}
+
+model JoinGroup {
```
Comment (qwerzl) [inline db/schema.prisma]: rename to `GroupInfo`

File: server/api/cas/group/club.get.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { PrismaClient } from '@prisma/client'
+import * as z from 'zod'
+
+const prisma = new PrismaClient()
+
+const requestSchema = z.object({
+  club: z.string(),
```
Comment (qwerzl) [inline server/api/cas/group/club.get.ts]: z.number()

File: server/api/cas/group/club.get.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  if (!joinGroup) {
+    return {
+      data: null,
+      status: 'No joinGroup information found for the club',
+    }
+  }
+
+  return {
+    data: joinGroup,
+    status: 'success',
```
Comment (qwerzl) [inline server/api/cas/group/club.get.ts]: I don't think we need to return this status

File: server/api/cas/group/delete.delete.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if (!auth.userId) {
+    setResponseStatus(event, 403)
+    return
+  }
+
+  const tsimsStudentId = await prisma.user.findUnique({
+    where: {
+      id: auth.userId,
+    },
+  }).then(user => user?.tsimsStudentId)
```
Comment (qwerzl) [inline server/api/cas/group/delete.delete.ts]:
```
(await prisma.user.findUnique({
    where: {
      id: auth.userId,
    },
  })).tsimsStudentId
```

File: server/api/cas/group/club.get.ts
Comment (qwerzl) [inline server/api/cas/group/club.get.ts:1]: rename to /api/cas/info/get.get.ts

File: server/api/cas/group/delete.delete.ts
Comment (qwerzl) [inline server/api/cas/group/delete.delete.ts:1]: Change to /api/cas/info/delete.delete.ts

Comment (qwerzl) [reply server/api/cas/group/delete.delete.ts:1]: I don't think there is a use case for this `delete` endpoint.

File: server/api/cas/group/new.post.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          presidentByTsimsStudentId: tsimsStudentId,
+        },
+        {
+          vicesByTsimsStudentId: {
+            has: tsimsStudentId,
+          },
+        },
+      ],
+    },
+  }).then(club => club !== null)
```
Comment (qwerzl) [inline server/api/cas/group/new.post.ts]: Just find a club which has the given id and `presidentByTsimsStudentId === tsimsStudentId`.

Comment (qwerzl) [reply server/api/cas/group/new.post.ts]: If undefined, return 403.

File: server/api/cas/group/update.post.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          presidentByTsimsStudentId: tsimsStudentId,
+        },
+        {
+          vicesByTsimsStudentId: {
+            has: tsimsStudentId,
+          },
+        },
+      ],
+    },
+  }).then(club => club !== null)
```
Comment (qwerzl) [inline server/api/cas/group/update.post.ts]: As above

File: types/api/cas/group/club.d.ts
Comment (qwerzl) [inline types/api/cas/group/club.d.ts:1]: I don't think we currently need this. The types of returned data from these endpoints are already given by prisma.

File: server/api/cas/group/club.get.ts (hunk 3 above)
Comment (qwerzl) [reply server/api/cas/group/club.get.ts]: Just return `joinGroup`, or `clubInfo` after the rename.

**Review by qwerzl — CHANGES_REQUESTED** [review]

**Pull request merged**